### DISCUSSION TITLE
CMAKE_CXX_SCAN_FOR_MODULES in vcpkg_cmake_configure()

### DISCUSSION THREAD

**Discussion by silverqx** [microsoft/vcpkg#38325] [General]
I want to ask if I can use the `CMAKE_CXX_SCAN_FOR_MODULES=OFF` during `vcpkg_cmake_configure()` call, something like:

```cmake
vcpkg_cmake_configure(
    SOURCE_PATH "${SOURCE_PATH}"
    OPTIONS
        -DCMAKE_CXX_SCAN_FOR_MODULES:BOOL=OFF
        -DBUILD_TESTS:BOOL=OFF
        -DBUILD_TREE_DEPLOY:BOOL=OFF
        ...
        ${FEATURE_OPTIONS}
)
```

**Comment by Neumann-A** [comment]
You probably want to use VCPKG_CMAKE_CONFIGURE_OPTIONS in a triplet to set it globally.

**Comment by silverqx** [comment]
It looks like there is no difference in setting it to `VCPKG_CMAKE_CONFIGURE_OPTIONS`, `vcpkg_cmake_configure()` only appends the values from `VCPKG_CMAKE_CONFIGURE_OPTIONS` back to `arg_OPTIONS` so it's unnecessary.

It looks like the `VCPKG_CMAKE_CONFIGURE_OPTIONS` is supposed to be used from triplets where it makes sense.

I need to know something a little different, if I disable the `CMAKE_CXX_SCAN_FOR_MODULES=OFF` if it will not interfere with something else, eg. if vcpkg doesn't internally use some code with modules that is added to my port build. I'm at 100% I'm not using C++ modules in my port.

Thx for advice @Neumann-A

**Reply by Neumann-A** [reply]
> vcpkg doesn't internally use some code with modules that is added to my port build

vcpkg does not add anything to building a specific port. It just invokes whatever upstream wants.

**Reply by silverqx** [reply]
That are good news, I searched all ports for `CMAKE_CXX_SCAN_FOR_MODULES` and even one uses it that is why I asked. It looks like it's safe to define it, I leave it in portfile for now. Thx
